mne_bids/write.py

```diff
@@ -858,13 +860,36 @@ def write_raw_bids(raw, bids_path, events_data=None,
         object, as well as the extension. Data with MEG and other
         electrophysiology data in the same file will be stored as ``'meg'``.
     events_data : path-like | array | None
+        Use this parameter to specify events to write to the ``*_events.tsv``
+        sidecar file, additionally to the object's `mne.Annotations` (which
+        are always written).
         If a path, specifies the location of an MNE events file.
         If an array, the MNE events array (shape: ``(n_events, 3)``).
-        If ``None``, events will be inferred from the the raw object's
-        `mne.Annotations` using `mne.events_from_annotations`.
+        If a path or an array and ``raw.annotations`` exist, the union of
+        ``event_data`` and ``raw.annotations`` will be written.
+        Corresponding descriptions for all event IDs (listed in the third
+        column of the MNE events array) must be specified via the ``event_id``
+        parameter; otherwise, an exception is raised.
+        If ``None``, events will only be inferred from the the raw object's
+        `mne.Annotations`.
+
+        .. note::
+           If ``not None``, writes the union of ``events_data`` and
+           ``raw.annotations``. If you wish to **only** write
+           ``raw.annotations``, pass ``events_data=None``. If you want to
+           **exclude** the events in ``raw.annotations`` from being written,
+           call ``raw.set_annotations(None)`` before invoking this function.
+
+        .. note::
+           Descriptions of all event IDs must be specified via the ``event_id``
```

Comment:
this does not include the annotations right? I think this will need a proper tutorial to explain all these different possibilities

Reply:
Annotations don't have event IDs, so yes, this only applies to events :)